src/generateRenderer/colorRamps.js

```diff
@@ -0,0 +1,54 @@
+const chroma = require('chroma-js')
+
+module.exports = { createMultipartRamp, createAlgorithmicRamp }
+
+/**
+*
+* generate multipart color ramp
+*
+* @param {array} options
+* @return {array} algorithmic colorRamps
+*/
+function createMultipartRamp (options) {
+  const { rampDetails, breakCount = 7 } = options
+  const type = rampDetails.type
+  const colorRamps = rampDetails.colorRamps
+
+  if (!type === 'multipart' && !colorRamps.length >= 1) return
```

Comment:
sounds good. I still need to run tests on multipart ramps. This may change again.